docs/proposals/20200730-ConfigV1Alpha2.md

```diff
@@ -0,0 +1,194 @@
+# Proposal: Config v1alpha2
+
+Author(s): Andrew Rynhard
+
+## Abstract
+
+The goal in this proposal is to outline the v1alpha2 config manifests and to detail how multi-doc YAML enables API and config driven management of Talos without introducing accidental complexity.
+
+## Background
+
+The `v1alpha1` config has limitations, and layout flaws that we would like to fix.
+In our experience with `v1alpha1` we found that a single configuration file made it difficult to evlove configuration options without breaking things.
+
+## Proposal
+
+```yaml
+---
+# Controlplane contains configuration options for the Talos and Kubernetes
+# control planes. The absence of this manifest enables automatic retrieval from
+# the Talos control plane.
+#
+# Dynamic: true
+# Required: false
+kind: controlplane
+version: v1alpha1
+spec:
+  talos:
+    ca:
+      key: ""
+---
+# Join contains the parameters required to join the Talos and Kubernetes control
+# planes.
+#
+# Required: true
+# Dynamic: false
+kind: join
+version: v1alpha1
+spec:
+  kubernetes:
+    endpoint: https://1.2.3.4
+    token: 5dt69c.npg6duv71zwqhzbg
+    ca:
+      crt: ""
+  talos:
+    token: 5dt69c.npg6duv71zwqhzbg
+    ca: # Client needs this to validate trustd connection.
+      crt: ""
+---
+# Etcd contains configuration options for the etcd service. The presence of
+# this manifest enables the etcd service.
+#
+# Required: true
+# Dynamic: true
+kind: etcd
+version: v1alpha1
+spec: {}
+---
+# Kubelet contains configuration options for the kubelet service. In the absence
+# of this manifest, a node will pull this manifest from the Talos control plane.
+#
+# Required: true
+# Dynamic: true
+kind: kubelet
+version: v1alpha1
+spec:
+  image: docker.io/autonomy/kubelet:latest
+  args: {}
+---
+# Machine contains machine configuration options.
+#
+# Required: true
+# Dynamic: false
+kind: machine
+version: v1alpha1
+spec:
+  identity:
+    sans: []
+  bootloader:
+    cmdline: []
+  kernel:
+    parameters: {}
+  storage: {}
+  time:
+    ntp:
+      servers: {}
+  env: {}
+  registries: {}
+---
+# Network contains network configuration options.
+#
+# Required: false
+# Dynamic: false
+kind: network
+version: v1alpha1
+spec:
+  interfaces:
+    - name: ""
+      dhcp: true
+      cidrs:
+        - 1.2.3.4/8
+  routes:
+    - destination:
+      gateway:
```

Comment:
@Ulexus In looking at how systemd does it, a route definition is required of dhcp is not enabled. Which makes it seem like `routes` should be part of the `interface` struct. Alternatively, I guess what I am missing from this `route` struct is a reference to the interface.

Reply:
No, I route definition is never required, unless I'm really missing something.  However, yes, routes should be bound to the interface.

Reply:
I very well could be misreading it. I am looking at https://wiki.archlinux.org/index.php/Systemd-networkd#Configuration_examples. And it says 
> Gateway= this option is mandatory unless DHCP is used
Under `[Route]`.

Reply:
Meaning that for each route statement, you have to have a gateway for that route.  That doesn't mean that you must have a route statement.

```
ip route add 192.168.2.0/24 via 192.168.1.1
```

In the iproute2 statement above, the `Destination` is `192.168.2.0/24` and the `Gateway` is `192.168.1.1`.

Reply:
:+1: I see now. Not sure how I missed that.

Reply:
The official documentation is better for a complete reference.  The arch documentation is excellent for getting your bearings, but the details are in the official freedesktop documentation.